File: pset1/credit.c
#include <stdio.h>
#include <cs50.h>

int main(void)
{
    printf("Number: \n");
    long long cc_number = get_long_long();
    int fst_sum = 0;
    int snd_sum = cc_number % 10;
    
    bool toggle = true;
    for (long long i=100; i< cc_number*10; i = i * 10) {
        int multip_digits = ((cc_number % i) / (i/10))*2;
        int other_numbers = (cc_number%i) /(i/10);
        
        if (toggle == true) {
            if (multip_digits > 9) {
                fst_sum += (multip_digits/10) + (multip_digits%10);
            } else {
                fst_sum += multip_digits;
            }
            toggle = false;
        } else if (toggle == false) {
            snd_sum += other_numbers;
            toggle = true;
        }
    }
    
    int test_number = fst_sum + snd_sum;
    if (test_number%10 == 0) {
        if (cc_number >= 5100000000000000 && cc_number < 5600000000000000) {
            printf("MASTERCARD\n");
        } else if (cc_number >= 340000000000000 && cc_number < 380000000000000) {
            printf("AMEX\n");
        } else {
            printf("VISA\n");
        }
    } else {
        printf("INVALID\n");
    }
}

File: pset6/credit.py

cc_input = "";
while True:
    print("Number: ", end="")
    cc_input = input()
    if cc_input.isdigit() == True:
        break

lsDigits = []
fst_sum = 0;
snd_sum = 0;

for digit in cc_input:
    lsDigits.append(int(digit))

for i in range(len(lsDigits)-2, -1, -2):#Iterating and multipling evry other digit by2 starting w/ the number's secnd-to-last digit
    if lsDigits[i] * 2 < 9:
        fst_sum += lsDigits[i] * 2
    else: #if the resulting product is greater than 9 add both digits to the sum (The product of a single digit by2 is always < 19)
        fst_sum += ((lsDigits[i] * 2) % 10 ) + 1

for i in range(len(lsDigits)-1, -1, -2):#Iterating and multipling every other digit by2 starting with the last number's digit
    snd_sum += lsDigits[i]

if (fst_sum + snd_sum) % 10 == 0: #Check if the credit card number is valid and, if True, check which company issues it
    if lsDigits[0] == 5:
        print("MASTERCARD")
    elif lsDigits[0] == 4:
        print("VISA")
    else:
        print("AMEX")
else:
    print("INVALID")

